### PR TITLE
Improve layout of `GridPlot` with expanding children

### LINKED ISSUE #12614: [BUG] sizing_mode='stretch_both' broken in gridplot (bokeh 3.0.2)
### Software versions

Python version      :  3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:12:32) [MSC v.1929 64 bit (AMD64)]
IPython version     :  (not installed)
Tornado version     :  6.2
Bokeh version       :  3.0.2
BokehJS static path :  C:\Users\***\anaconda3\envs\bokeh3\Lib\site-packages\bokeh\server\static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  Windows-10-10.0.22000-SP0

### Browser name and version

Chrome, Firefox, Edge

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

Plots arranged in a gridplot are not shown correctly, when compared to bokeh 2.4.3

I simplified this example:
https://docs.bokeh.org/en/3.0.2/docs/examples/basic/data/color_mappers.html

With bokeh 2.4.3 it shows 4 plots that stretch to the edges of the browser.

### Observed behavior

With bokeh 3.0.1 and 3.0.2 the plot is messed up. The toolbar is in the correct place, but apart from some fragments of the axis, the plots are too small to be visible. They appear compressed to 0 width and height in the top left corner.

### Example code

```Python
import numpy as np

from bokeh.layouts import gridplot
from bokeh.models import ColumnDataSource
from bokeh.plotting import figure, show

x = np.random.random(size=2000) * 1000
y = np.random.normal(size=2000) * 2 + 5
source = ColumnDataSource(dict(x=x, y=y))

def make_plot(mapper_type, palette):
    p = figure(x_range=(1, 1000), title=f"{palette} with {mapper_type} mapping",
               x_axis_type=mapper_type)
    p.scatter('x', 'y', alpha=0.8, source=source)
    return p

p1 = make_plot("linear", "Viridis256")
p2 = make_plot("log", "Viridis256")
p3 = make_plot("linear", "Viridis6")
p4 = make_plot("log", "Viridis6")

grid =  gridplot([[p1, p2], [p3, p4]],
                sizing_mode='stretch_both',
                 # width=400, height=300,
                 # toolbar_location=None,
                 )
show(grid)
```

### Stack traceback or browser console output

browser console does not seem to show errors

### Screenshots

image

### LINKED ISSUE #12863: [BUG] GridPlot sizing mode not respected
### Software versions

Bokeh: 3.1.0dev3

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

I expected plots with `stretch_width` or `stretch_both` sizing modes to fill the entire area of a `GridPlot`. 

### Observed behavior

The plots only fill the `min_width`.

### Example code

```Python
from bokeh.io import curdoc
from bokeh.plotting import figure
from bokeh.layouts import gridplot

plots = [figure(sizing_mode='stretch_width', min_width=300, height=300) for _ in range(4)]

for p in plots:
    p.scatter([1, 2, 3], [1, 2, 3])
    
g = gridplot([plots[:2], plots[2:]], sizing_mode='stretch_width')

curdoc().add_root(g)
```

### Stack traceback or browser console output

_No response_

### Screenshots

Screen Shot 2023-02-27 at 19 29 19

### PR DESCRIPTION
**Pull request by mattpap**
This replaces the previous default of either `1fr` or `max-content` (depending on the context) with `auto` track sizing, which can cover both cases of fitting track width or height to track children dimensions, and allowing expansion of children when expanding sizing modes are involved. I also improved types and documentation of properties defining grid tracks.

fixes #12863
fixes #12614

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap can you add 1-2 sentences on PRs like this that describe the problem/fix at a very high level? You're currently the only real expert on layout so it would be useful for anyone looking at this now or in the future.

File: src/bokeh/models/plots.py [hunk 1]
```diff
@@ -880,11 +880,11 @@ def __init__(self, *args, **kwargs) -> None:
...
 
-    rows = Nullable(TracksSizing, default="max-content", help="""
+    rows = Nullable(TracksSizing, default=None, help="""
     Describes how the grid should maintain its rows' heights.
     """)
 
-    cols = Nullable(TracksSizing, default="max-content", help="""
+    cols = Nullable(TracksSizing, default=None, help="""
     Describes how the grid should maintain its columns' widths.
     """)
```
Comment (kernc) [inline src/bokeh/models/plots.py]: Curious, what are some possible values for these properties? The descriptions I find scarce of meaning.
https://github.com/bokeh/bokeh/blob/fbba86223a7d4c8415e16319fe3c930ee91b3c33/src/bokeh/models/layouts.py#L358

Comment (mattpap) [reply src/bokeh/models/plots.py]: I updated the types to be more specific and match bokehjs' types, and added more useful docstrings.

File: src/bokeh/core/property_aliases.py [hunk 2]
```diff
@@ -88,6 +92,18 @@
     )
 )
 
+GridSpacing = Either(Pixels, Tuple(Pixels, Pixels))
+
+TrackAlign = Enum("start", "center", "end", "auto")
+
+# CSS length, percentage, flex, max-content, min-content, auto, etc.
+# See https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns.
+TrackSize = String()
```
Comment (mattpap) [inline src/bokeh/core/property_aliases.py:101]: Later I will implement types to cover what's listed in the comment.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
I didn't really take a close look but approved on the basis of added test

**Pull request merged**